sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -2831,3 +2831,63 @@ async def test_state_deferred_async(self, uamqp_transport, *, servicebus_namespa
+        async with ServiceBusClient.from_connection_string(
+            servicebus_namespace_connection_string, uamqp_transport=uamqp_transport) as sb_client:
+
+            sender = sb_client.get_queue_sender(servicebus_queue.name)
+            for i in range(10):
+                message = ServiceBusMessage("message no. {}".format(i))
+                await sender.send_messages(message)
+
+            receiver = sb_client.get_queue_receiver(servicebus_queue.name, receive_mode=ServiceBusReceiveMode.RECEIVE_AND_DELETE)
+            number_deleted_messages = 0
+            async with receiver:
+                number_deleted_messages = await receiver.delete_messages(max_message_count=10, before_enqueued_time_utc=datetime.utcnow())
+            assert number_deleted_messages == 10
+
+            receiver_peek = sb_client.get_queue_receiver(servicebus_queue.name)
+            async with receiver_peek:
+                assert await receiver_peek.delete_messages() == 0
+
+            receiver = sb_client.get_queue_receiver(servicebus_queue.name, receive_mode=ServiceBusReceiveMode.RECEIVE_AND_DELETE)
+            async with receiver:
+                assert await receiver_peek.delete_messages() == 0
+
+    @pytest.mark.asyncio
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedServiceBusResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    @ServiceBusQueuePreparer(name_prefix='servicebustest', dead_lettering_on_message_expiration=True)
+    @pytest.mark.parametrize("uamqp_transport", uamqp_transport_params, ids=uamqp_transport_ids)
+    @ArgPasserAsync()
+    async def test_message_deleted_maximum_async(self, uamqp_transport, *, servicebus_namespace_connection_string=None, servicebus_queue=None, **kwargs):
+        with ServiceBusClient.from_connection_string(
+            servicebus_namespace_connection_string, uamqp_transport=uamqp_transport) as sb_client:
+
+            sender = sb_client.get_queue_sender(servicebus_queue.name)
+            batch_message = await sender.create_message_batch()
+            for _ in range(4001):
+                try:
+                    batch_message.add_message(ServiceBusMessage("Message inside a ServiceBusMessageBatch"))
+                except ValueError:
+                    # ServiceBusMessageBatch object reaches max_size.
+                    # New ServiceBusMessageBatch object can be created here to send more data.
+                    break
+            await sender.send_messages(batch_message)
+
+            receiver = sb_client.get_queue_receiver(servicebus_queue.name)
+            number_deleted_messages = 0
+            async with receiver:
+                number_deleted_messages = await receiver.delete_messages(max_message_count=4000, before_enqueued_time_utc=datetime.utcnow())
+            assert number_deleted_messages == 4000
```

Comment:
would we want to test deleting all messages in the queue when at least a few of them are locked? or not, since it won't necessarily change behavior on the client side?